plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/graphvisitors/InteractiveLayeredGraphVisitor.java

```diff
@@ -0,0 +1,451 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Kiel University and others.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0  *******************************************************************************/
+package org.eclipse.elk.alg.layered.graphvisitors;
```

Comment:
Since we won't have many of these at all, I'd simply put it into the `graph` package.

Reply:
One could argue that `graph` is only about the structure. Put it in the root package?

Reply:
I think for specific algorithms one might need a non interactive layout configurator which configures the "normal" layout before the interactive layout. But I am not sure whether this is necessary. I guess the root package seems appropriate.

Reply:
I proposed to put it in the `graph` package because it deals directly with the ELK graph instead of with our internal data structure. Confining ELK graph-specific code to a specific sub package makes it easier to remove ELK graph dependencies from ELK Layered.

However, we already have such a dependency in the root package (`LayeredLayoutProvider`), so my point is probably moot anyway. I'm fine with the root package.

> I think for specific algorithms one might need a non interactive layout configurator which configures the "normal" layout before the interactive layout.

@soerendomroes Why?

Reply:
For example one wants to set specific strategies or properties for example separate connected components to false for the layered approach. This property hinders the algorithm to infer the correct layers.